ironfist/src/cpp/adventure/adv.cpp

```diff
@@ -2,6 +2,57 @@
 #include "adventure/map.h"
 #include "game/game.h"
 #include "scripting/hook.h"
+#include "prefs.h"
+
+extern void __fastcall NormalDialog(char *, int, int, int, int, int, int, int, int, int);
+
+int advManager::ProcessDeSelect(struct tag_message *GUIMessage_evt, int *a3, class mapCell * *a4) {
+	if(GUIMessage_evt->yCoordOrFieldID == 4)
+		{
+		DWORD hero_reminder_reg_dword = read_pref<DWORD>("Show Hero Movement Reminder");
+		//default is true, but read_pref() returns -1 if the value is not set
+		bool show_hero_movement_reminder = !(hero_reminder_reg_dword == 0);
+		
+		if(!show_hero_movement_reminder || !gpCurPlayer->HasMobileHero() ||
+				(
+				NormalDialog("One or more heroes may still move, are you sure you want to end your turn?",
+							2, -1, -1, -1, 0, -1, 0, -1, 0),
+				gpWindowManager->buttonPressedCode != 30726)

```

Comment:
"30726" is another example of a magic number.  You can figure out what it means by using h2guiview on whatever .bin file the NormalDialog function opens. Certainly, it's hard to know what it means just by seeing it.

We still don't have a great way of dealing with constants in the .bin GUI files, but I think we do have a few examples of doing this reasonably -- I think displaying the mage guild is one.

Yes, Tales of Enroth included a similar magic constant, and I let it into Ironfist. Shame on me.